packages/flutter/lib/src/services/platform_channel.dart

```diff
@@ -142,8 +142,28 @@ class MethodChannel {
   BinaryMessenger get binaryMessenger => _binaryMessenger ?? ServicesBinding.instance!.defaultBinaryMessenger;
   final BinaryMessenger? _binaryMessenger;
 
+  /// Backend implementation of [invokeMethod].
+  ///
+  /// The `method` and `arguments` arguments are used to create a [MethodCall]
+  /// object that is passed to the [codec]'s [MethodCodec.encodeMethodCall]
+  /// method. The resulting message is then sent to the embedding using the
+  /// [binaryMessenger]'s [BinaryMessenger.send] method.
+  ///
+  /// If the result is null and `missingOk` is true, this returns null. (This is
+  /// the behaviour of [OptionalMethodChannel.invokeMethod].)
+  ///
+  /// If the result is null and `missingOk` is false, this throwsa
+  /// [MissingPluginException]. (This is the behaviour of
+  /// [MethodChannel.invokeMethod].)
+  ///
+  /// Otherwise, the result is decoded using the [codec]'s
+  /// [MethodCodec.decodeEnvelope] method.
+  ///
+  /// The `T` type argument is the expected return type. It is treated as
+  /// nullable.
+  @protected
   @optionalTypeArgs
-  Future<T?> _invokeMethod<T>(String method, { required bool missingOk, dynamic arguments }) async {
+  Future<T?> sendMethodCall<T>(String method, { required bool missingOk, dynamic arguments }) async {
```

Comment:
This will mean you need to change L481 to use this method instead of the now missing private one.

Why take away the private implementation? This is probably fine if it doesn't break anyone cause they implement `MethodChannel`, but might be a pain to make a breaking change over.

Reply:
i want to override it in google3

Reply:
(where someone implements this class as a proxy and it's causing me trouble with my ChannelBuffers PR, and i want to move them to extending the class instead)